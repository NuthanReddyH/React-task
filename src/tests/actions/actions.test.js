import configureMockStore from 'redux-mock-store'

const addNewCompany = (data) => ({
  type: 'ADD_COMPANY',
  data
});

// Create a mock store
const mockStore = configureMockStore()
const store = mockStore({})
describe('action creators', () => {
  it('creates ADD_COMPANY when adding a company was successful', () => {
    store.dispatch(addNewCompany(
      {
        companyName: 'Amazon',
        address: '410 Terry Ave Seattle,WA',
        revenue: '100000',
        employees: [],
        phone: '(022)-266-0000'
      }
    ));
    expect(store.getActions()).toMatchSnapshot();
  });
});
