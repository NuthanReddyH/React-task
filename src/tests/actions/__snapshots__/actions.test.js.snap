// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action creators creates ADD_COMPANY when adding a company was successful 1`] = `
Array [
  Object {
    "data": Object {
      "address": "410 Terry Ave Seattle,WA",
      "companyName": "Amazon",
      "employees": Array [],
      "phone": "(022)-266-0000",
      "revenue": "100000",
    },
    "type": "ADD_COMPANY",
  },
]
`;

exports[`action creators creates ADD_COMPANY when creating a to-do was successful 1`] = `
Array [
  Object {
    "data": Object {
      "address": "410 Terry Ave Seattle,WA",
      "companyName": "Amazon",
      "employees": Array [],
      "phone": "(022)-266-0000",
      "revenue": "100000",
    },
    "type": "ADD_COMPANY",
  },
]
`;

exports[`action creators dispatches correct action and payload 1`] = `
Array [
  Object {
    "data": Object {
      "address": "410 Terry Ave Seattle,WA",
      "companyName": "Amazon",
      "employees": Array [],
      "phone": "(022)-266-0000",
      "revenue": "100000",
    },
    "type": "ADD_COMPANY",
  },
  Object {
    "data": Object {
      "address": "410 Terry Ave Seattle,WA",
      "companyName": "Amazon",
      "employees": Array [],
      "phone": "(022)-266-0000",
      "revenue": "100000",
    },
    "type": "ADD_COMPANY",
  },
]
`;
